src/mocap_pose.cpp

```diff
@@ -249,7 +258,15 @@ void MocapPose::WorkerThread()
 
         bool dataAvailable = false;
         bool streamFrames = false;
-        unsigned short udpPort = 6734;
+        // randomize port between 6734 and maximum port.
+        // this is because if we're behind a NAT gateway, the port cannot be ambiguous if there's multiple
+        // mocap-pose clients behind the same NAT gateway.
+        unsigned short udpPort = 6734 + (rand() % (65535 - 6734));
```

Comment:
Please review this randomization logic carefully. I already had a math bug once in this line, so don't trust me.

Reply:
The randomization logic seems to be fine.
I am thinking that should this line be moved outside the while loop. I am thinking that we might want to have the same port on one session of the mocap_pose execution.
What do you think?

Reply:
Good thinking! Pros and cons of keeping the port the same across retries:

Pros
- QTM server has a change of associating the data stream with the previous connection, so it can clean up or consolidate resources

Cons
- QTM server could refuse to start data stream again for the same UDP port

But it can't hurt to try, and as it has always worked before for a static UDP port I guess that should be well supported. Good idea and thanks for the feedback.

Will fix.